src/Shared/Microsoft.DotNet.Internal.Testing.Utility/GenericReadOnlyDictionaryAssertions.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using FluentAssertions.Collections;
+
+namespace Microsoft.DotNet.Internal.Testing.Utility
+{
+    public class GenericReadOnlyDictionaryAssertions<TKey, TValue> : GenericDictionaryAssertions<TKey, TValue>
+    {
+        private class ReadOnlyDictionaryWrapper : IDictionary<TKey, TValue>
+        {
+            private readonly IReadOnlyDictionary<TKey, TValue> _backing;
+
+            public ReadOnlyDictionaryWrapper(IReadOnlyDictionary<TKey, TValue> backing)
+            {
+                _backing = backing;
+            }
+
+            public IEnumerator<KeyValuePair<TKey, TValue>> GetEnumerator()
+            {
+                return _backing.GetEnumerator();
+            }
+
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                return ((IEnumerable) _backing).GetEnumerator();
+            }
+
+            public void Add(KeyValuePair<TKey, TValue> item)
+            {
+                throw new NotSupportedException();
+            }
+
+            public void Clear()
+            {
+                throw new NotSupportedException();
+            }
+
+            public bool Contains(KeyValuePair<TKey, TValue> item)
+            {
+                throw new NotSupportedException();
```

Comment:
>throw new NotSupportedException(); [](start = 16, length = 34)

This should probably not be throwing. Contains is supported on Read Only things.

Reply:
It's not, actually. Which is... odd...  IReadOnlyDictionary doesn't have a Contains(KVP) method.

Reply:
Probably because it's provided by EnumerableExtensions

Reply:
Though it's possible fluent assertions will want it... I'll make it work.

Reply:
Added